tests/unit/AmalgamatingBusinesses.spec.ts

```diff
@@ -87,15 +93,368 @@ describe.skip('Amalgamating Businesses - add amalgamating business', () => {
+      legalType: CorpTypeCd.EXTRA_PRO_A,
+      name: 'Extra Pro Business',
+      identifier: 'A1234567'
+    })
+
+    // verify snackbar is displayed
+    expect(wrapper.vm.snackbar).toBe(true)
+    expect(wrapper.vm.snackbarText).toBe('Business is already in table.')
+
+    // verify data
+    expect(store.stateModel.amalgamation.amalgamatingBusinesses.length).toBe(1)
+
+    // verify panel is now closed
+    expect(wrapper.vm.isAddingAmalgamatingBusiness).toBe(false)
   })
 
-  it('saves an amalgamating business', async () => {
+  it('save an amalgamating business - BC', async () => {
+    // set state
+    store.setCurrentDate('2023-01-01')
+    await Vue.nextTick()
+
+    // open panel
+    await wrapper.setData({ isAddingAmalgamatingBusiness: true })
+
+    // mock services functions
+    vi.spyOn((AuthServices as any), 'fetchAuthInfo').mockImplementation(() => ({
+      contacts: [{ email: 'email@example.com' }]
+    }))
+    vi.spyOn((LegalServices as any), 'fetchBusinessInfo').mockImplementation(() => ({
+      identifier: 'BC1234567',
+      legalName: 'My BC Business',
+      legalType: CorpTypeCd.BC_COMPANY,
+      goodStanding: true,
+      adminFreeze: false,
+      state: EntityStates.ACTIVE,
+      stateFiling: 'mock_url',
+      restorationExpiryDate: '2023-01-18'
+    }))
+    vi.spyOn((LegalServices as any), 'fetchAddresses').mockImplementation(() => ({
+      registeredOffice: { mailingAddress: { streetAddress: '123 Main St' } }
+    }))
+    vi.spyOn((LegalServices as any), 'fetchFirstOrOnlyFiling').mockImplementation(() => ({
+      isFutureEffective: false,
+      status: FilingStatus.COMPLETED
+    }))
+    // in this case, return a promise so that ".catch()" is defined on the implementation
+    vi.spyOn((LegalServices as any), 'fetchFiling').mockImplementation(() =>
+      Promise.resolve({
+        restoration: { type: RestorationTypes.FULL }
+      })
+    )
+
+    // simulate saving a BC business
+    await wrapper.vm.saveAmalgamatingBusiness({
+      legalType: CorpTypeCd.BC_COMPANY,
+      name: 'My BC Business',
+      identifier: 'BC1234567'
+    })
+
+    // verify data
+    expect(store.stateModel.amalgamation.amalgamatingBusinesses.length).toBe(1)
+    const business = store.stateModel.amalgamation.amalgamatingBusinesses[0] as any
+    expect(business.type).toBe(AmlTypes.LEAR)
+    expect(business.role).toBe(AmlRoles.AMALGAMATING)
+    expect(business.identifier).toBe('BC1234567')
+    expect(business.name).toBe('My BC Business')
+    expect(business.email).toBe('email@example.com')
+    expect(business.legalType).toBe(CorpTypeCd.BC_COMPANY)
+    expect(business.address.streetAddress).toBe('123 Main St')
+    expect(business.isNotInGoodStanding).toBe(false)
+    expect(business.isFrozen).toBe(false)
+    expect(business.isFutureEffective).toBe(false)
+    expect(business.isPendingFiling).toBe(false)
+    expect(business.isLimitedRestoration).toBe(false)
+    expect(business.isHistorical).toBe(false)
```

Comment:
In retrospect, I could have checked if `business` was "equal to" an object with all these properties.